pkg/providers/ovirt/client/ovirt-client.go

```diff
@@ -0,0 +1,289 @@
+
+			followed, err := client.connection.FollowLink(dogs)
+			if err != nil {
+				return err
+			}
+			vm.SetWatchdogs(followed.(*ovirtsdk.WatchdogSlice))
+		}
+	}
+	return nil
+}
+
+func (client *richOvirtClient) populateGrapthicsConsoles(vm *ovirtsdk.Vm) error {
+	if consoles, ok := vm.GraphicsConsoles(); ok {
+		if _, ok := consoles.Href(); ok {
+			followed, err := client.connection.FollowLink(consoles)
+			if err != nil {
+				return err
+			}
+			vm.SetGraphicsConsoles(followed.(*ovirtsdk.GraphicsConsoleSlice))
+		}
+	}
+	return nil
+}
+
+func (client *richOvirtClient) populateQuota(vm *ovirtsdk.Vm) error {
+	if quota, ok := vm.Quota(); ok {
+		if _, ok := quota.Href(); ok {
+			followed, err := client.connection.FollowLink(quota)
+			if err != nil {
+				return err
+			}
+			vm.SetQuota(followed.(*ovirtsdk.Quota))
+		}
+	}
+	return nil
+}
+
+func (client *richOvirtClient) populateNics(vm *ovirtsdk.Vm) error {
+	if nics, ok := vm.Nics(); ok {
+		if _, ok := nics.Href(); ok {
+			followed, err := client.connection.FollowLink(nics)
+			nics = followed.(*ovirtsdk.NicSlice)
+			if err != nil {
+				return err
+			}
+			for _, nic := range nics.Slice() {
+				if prof, ok := nic.VnicProfile(); ok {
+					if _, ok := prof.Href(); ok {
+						vnicFollowed, err := client.connection.FollowLink(prof)
+						if err != nil {
+							return err
+						}
+						vnic := vnicFollowed.(*ovirtsdk.VnicProfile)
+						if net, ok := vnic.Network(); ok {
+							if _, ok := net.Href(); ok {
+
+								network, err := client.connection.FollowLink(net)
+								if err != nil {
+									return err
+								}
+								vnic.SetNetwork(network.(*ovirtsdk.Network))
+							}
+						}
+						nic.SetVnicProfile(vnic)
+					}
+				}
+				if net, ok := nic.Network(); ok {
+					if _, ok := net.Href(); ok {
+
+						network, err := client.connection.FollowLink(net)
+						if err != nil {
+							return err
+						}
+						nic.SetNetwork(network.(*ovirtsdk.Network))
+					}
+				}
+			}
+			vm.SetNics(nics)
+		}
+	}
+	return nil
+}
+
+func (client *richOvirtClient) populateDiskAttachments(vm *ovirtsdk.Vm) error {
+	if das, ok := vm.DiskAttachments(); ok {
+		if _, ok := das.Href(); ok {
+
+			followed, err := client.connection.FollowLink(das)
+			if err != nil {
+				return err
+			}
+			attachments := followed.(*ovirtsdk.DiskAttachmentSlice)
+			for _, da := range attachments.Slice() {
+				if disk, ok := da.Disk(); ok {
+					if _, ok := disk.Href(); ok {
+
+						diskFollowed, err := client.connection.FollowLink(disk)
+						if err != nil {
+							return err
+						}
+						da.SetDisk(diskFollowed.(*ovirtsdk.Disk))
+					}
+				}
+			}
+			vm.SetDiskAttachments(attachments)
+		}
+	}
+	return nil
+}
+
+func connect(apiURL string, username string, password string, caFile string, insecure bool) (*ovirtsdk.Connection, error) {
+	connection, err := ovirtsdk.NewConnectionBuilder().
+		URL(apiURL).
+		Username(username).
+		Password(password).
+		CAFile(caFile).
```

Comment:
When we have a ca we do not need to set the insecure flag. Do we expect second secret to be mounted as a file?

Reply:
Yes, the CA file should be made available this way or downloaded by the client and put in /tmp.
For testing purposes I think it's useful to still be able to set the connection to insecure.